### PR TITLE
Add the option to select the display of more documents from folders on the same page #109

### PR DESCRIPTION
**Pull request by KebabRonin**
* Made the query string impact the amount of files shown
* Added an option select for files per page

image

### CODE REVIEW

File: ui/src/main/resources/FileManagerCode/Pagination.xml [hunk 1]
```diff
@@ -213,6 +220,11 @@
           scope.offset = pageNumber * scope.limit;
         };
 
+        scope.onChangedLimitOption = function() {
+          scope.limit=this.limit;
+          $location.search('filesLimit', scope.limit);
```
Comment (KebabRonin) [inline ui/src/main/resources/FileManagerCode/Pagination.xml:225]: When updating the query string, the live table is updated but the last page number of the page list is not updated.

File: ui/src/main/resources/FileManagerCode/Pagination.xml [hunk 2]
```diff
@@ -56,6 +56,16 @@
       &lt;/span&gt;
     &lt;/span&gt;
 
+    &lt;span ng-if="showResultsRange &amp;&amp; totalCount &gt; 0" class='resultsNo xwiki-livetable-pagesize'&gt;
+      &lt;span&gt;$services.localization.render('platform.livetable.pagesizeLabel')&lt;/span&gt;
```
Comment (mflorea) [inline ui/src/main/resources/FileManagerCode/Pagination.xml:60]:
```suggestion
      &lt;span&gt;$escapetool.xml($services.localization.render('platform.livetable.pagesizeLabel'))&lt;/span&gt;
```

File: ui/src/main/resources/FileManagerCode/Pagination.xml [hunk 3]
```diff
@@ -213,6 +223,11 @@
           scope.offset = pageNumber * scope.limit;
         };
 
+        scope.onChangedLimitOption = function() {
+          scope.limit=this.limit;
```
Comment (mflorea) [inline ui/src/main/resources/FileManagerCode/Pagination.xml:227]:
```suggestion
          scope.limit = this.limit;
```

File: ui/src/main/resources/FileManagerCode/Pagination.xml [hunk 4]
```diff
@@ -223,6 +238,15 @@
           var offset = newValues[1] || 0;
           var limit = newValues[2] || 15;
 
+          if (!scope.filesLimitOptions &amp;&amp; newValues[2]) {
```
Comment (mflorea) [inline ui/src/main/resources/FileManagerCode/Pagination.xml:241]: What if ``filesLimitOptions`` is defined, but the specified limit is not included in the list?

Comment (KebabRonin) [reply ui/src/main/resources/FileManagerCode/Pagination.xml:241]: The list will remain the same, but the filter will be applied in the LiveTable (the selected option will become blank).

On a new page load (with no query string), the `if` executes and adds 15 to the list. Afterwards, the only way to change the files per page to a custom amount is by modifying the query string (i.e. `l=44`, which isn't added to the list). Refreshing the page does add the new value to the options.

I initially added the `if` to not recompute the options every time one of `totalCount`, `offset` or `limit` has a value change.

File: ui/src/main/resources/FileManagerCode/Pagination.xml (hunk 2 above)
Comment (KebabRonin) [reply ui/src/main/resources/FileManagerCode/Pagination.xml:60]: Should I add escapes for other similar `$services.localisation.render` calls which were not added as part of this commit?

File: ui/src/main/resources/FileManagerCode/Pagination.xml [hunk 5]
```diff
@@ -224,10 +224,22 @@
...
           $location.search('filesLimit', scope.limit);
         };
 
+        scope.$watch('limit', function(newValue, oldValue) {
+          let limit = newValue;
+          if (!scope.filesLimitOptions) {
+            scope.filesLimitOptions = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100];
+          }
+          // If the page has a custom number of files per page, include it in the options.
+          if (typeof(limit) == 'number' &amp;&amp; !scope.filesLimitOptions.includes(limit)) {
```
Comment (mflorea) [inline ui/src/main/resources/FileManagerCode/Pagination.xml:237]:
```suggestion
          if (typeof(limit) === 'number' &amp;&amp; !scope.filesLimitOptions.includes(limit)) {
```

**Review by mflorea — COMMENTED** [review]
Looks good, with only a minor comment.

**Review by mflorea — APPROVED** [review]

**Pull request merged**